### PR TITLE
Use PasswordEncoder in OAuth2ClientAuthenticationProvider

### LINKED ISSUE #271: Use PasswordEncoder to verify client credentials
Now client credentials are kept in clear text and verified using simple string comparison. Provide possibility to inject proper PasswordEncoder implementation.

### PR DESCRIPTION
**Pull request by rlewczuk**
closes #271

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@rlewczuk Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects-experimental/spring-authorization-server&pullRequestId=272)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects-experimental/spring-authorization-server&pullRequestId=272) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@rlewczuk Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects-experimental/spring-authorization-server&pullRequestId=272)!

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java [hunk 1]
```diff
@@ -64,11 +66,13 @@
 	 * @param authorizationService the authorization service
 	 */
 	public OAuth2ClientAuthenticationProvider(RegisteredClientRepository registeredClientRepository,
-			OAuth2AuthorizationService authorizationService) {
+			OAuth2AuthorizationService authorizationService, PasswordEncoder passwordEncoder) {
```
Comment (jgrandja) [inline oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java:69]: Please revert constructor parameter and instead add `setPasswordEncoder()`. The default should be set using `PasswordEncoderFactories.createDelegatingPasswordEncoder()` and `setPasswordEncoder()` will only be called if a `@Bean` exists.

**Review by jgrandja — CHANGES_REQUESTED** [review]
Thanks for the PR @rlewczuk. Please see review comment.

File: oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java (hunk 1 above)
Comment (rlewczuk) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java]: Fixed.

Comment (rlewczuk) [reply oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java]: Hi @jgrandja. I'm marking this review comment as resolved, but I'm not sure how it works in Spring team. If reviewers should do it, let me know.

**Comment by jgrandja** [comment]
Thanks for the updates @rlewczuk. This is now merged.

FYI, I applied some updates with a polish commit.
Please let me know if you have any questions.
Thanks!

**Comment by rlewczuk** [comment]
Looked through polishing commit, there was more polishing than actual code.  Well, I'm still newcomer learning how things work here, so I'll try to be more careful in the future. Regards.

**Pull request closed without merging**